test/cppsuite/test_harness/workload_validation.h

```diff
@@ -203,10 +203,12 @@ class workload_validation {
         cursor->set_key(cursor, collection_name.c_str(), key_str.c_str());
         testutil_check(cursor->search_near(cursor, &exact));
         /*
-         * Since the timestamp which is part of the key is not provided, exact is expected to be
-         * greater than 0.
+         * Since the timestamp which is part of the key is not provided, exact cannot be 0. If it is
+         * -1, we need to go to the next key.
          */
-        testutil_assert(exact >= 0);
+        testutil_assert(exact != 0);
```

Comment:
What if its +1?

Reply:
Then we are on the right record.
As stated in the comment, the timestamp is not provided when search_near is called. The retrieved key will be bigger than the one that has been set, hence search_near returns 1.

Reply:
Can you please test this with a stressful configuration on our POC test which is I believe how we first found this bug?

Reply:
It was tested with:

`

# Sets up a basic database with 2 collections and 500 keys and run thread for 10 seconds.
--
# All components are enabled.
# Used as a basic test for the framework.
duration_seconds=10,
cache_size_mb=1000,
enable_logging=true,
runtime_monitor=
(
rate_per_second=3,
stat_cache_size=
(
enabled=true,
limit=100
)
),
timestamp_manager=
(
enabled=true,
oldest_lag=1,
stable_lag=1
),
workload_generator=
(
collection_count=2,
key_count=500,
key_size=10,
ops_per_transaction=
(
min=5,
max=50
),
read_threads=1,
update_threads=1,
value_size=20000
),
workload_tracking=
(
enabled=true
)

`

Reply:
Increase the duration to 120 and keycount to 50000, cache size MB can be like 5000 and value size 200000.

Reply:
With such a value size, we end up running out of memory. We will need to address this. I changed the value_size to be 2000 and it works.

Reply:
I have also noticed that if the duration_seconds parameter is not large enough (i.e the populate phase takes more time than the value attributed to duration_seconds), we run into an issue. I created WT-7446 to address this.

Reply:
That should only use 10GB of memory so I'm surprised its running out of memory, what machine are you running it on?

Reply:
It seems that we are allocating too much memory during the validation process. We could be more memory efficient during that step. In check_disk_state, verify_collection_state, is_key_present and verify_value, we are creating a new cursor and allocating memory to it.

Reply:
As discussed, a new stressful configuration will be added: config_poc_test_stress.txt. We will be able to make it more stressing once the memory issues are fixed.